ofctrl/fgraphFlow.go

```diff
@@ -38,7 +38,9 @@ type FlowMatch struct {
 	MacSa         *net.HardwareAddr    // Mac source
 	MacSaMask     *net.HardwareAddr    // Mac source mask
 	Ethertype     uint16               // Ethertype
-	VlanId        uint16               // vlan id
+	NonVlan       bool                 // Non-vlan
+	VlanId        *uint16              // vlan id
```

Comment:
This change is incompatible for those code which is using this library. Can we still keep VlanIdd uint16 and only add new fields to do new work? such as:
if NonVlan { // For those code using this library, it should be initialized as 0
  // new work code for vlanId = 0
} else if VlanId !=0 {
    if vlanMask != 0 {
    // new code change
    } else {
    // old compatible code
    }
}
Or other similar code logic.

Reply:
This is incompatible, but easy to fix due to compile check.
@wenyingd Do you think we should keep `uint16` for `VlanId`?

Reply:
If we use *uint16, we can support to match packet with 802.1q tag and vlanid=0.

Reply:
In my opinion, if we want to support VLAN==0 in function, changing `uint16` type is needed. @commandgjj Do you have other suggestions if we don't change type but also want to match VLAN==0?

Reply:
For a packet, it can only be with Vlan or NonVlan. We can use NonVlan flag to check vlan set or not. If NonVlan is false, we get vlan_id from VlanID field even it is 0. Maybe as follows:
if self.Match.NonVlan { // Non-VLAN case
    vidField := openflow13.NewVlanIdField(0, nil)
    vidField.Value = new(openflow13.VlanIdField)
    ofMatch.AddField(*vidField)
} else { // VLAN case
        vidField := openflow13.NewVlanIdField(self.Match.VlanId, self.Match.VlanMask)
        ofMatch.AddField(*vidField)
}

Please check above suggestion. You can continue to change the VlanId type to *uint16 if it doesn't work.

Reply:
@commandgjj The default behavior should be "no VlanIdField" for ofMatch, thus we cannot `get vlan_id from VlanID field even it is 0`.

Reply:
OK, I understand. We may only use *uint16 to do this work.